contracts/utils/cryptography/MessageHashUtils.sol

```diff
@@ -63,6 +63,21 @@ library MessageHashUtils {
         return keccak256(abi.encodePacked(hex"19_00", validator, data));
     }
 
+    /**
+     * @dev Variant of {toDataWithIntendedValidatorHash-address-bytes-} optimized for cases where `data` is a bytes32.
+     */
+    function toDataWithIntendedValidatorHash(
+        address validator,
+        bytes32 messageHash
+    ) internal pure returns (bytes32 digest) {
+        assembly ("memory-safe") {
+            mstore(0x00, hex"19_00")
+            mstore(0x02, shl(96, validator))
```

Comment:
TODO: sanitize address

Reply:
wdym?

Reply:
I added that note during a meeting. Someone brote that up. Often when we write memory in assembly we have so assume that the input values may contain dirty upper bits.

However now that I look at it more carefully, the `shl(96, ...)` part should already fix that. 

@ernestognw can you confirm its fine?

Reply:
The first part is not always true. Its quite easy to generate "dirty" addresses that have non-zero upper bits. The `shl` should indeed address that.

I'm adding fuzzing tests